build/mac/build_delta_installer.py

```diff
@@ -21,29 +21,47 @@ def Main(argv):
       type='string', default=None, help='The path of old app bundle.')
   parser.add_option('--new-app', dest='new_app_path', action='store',
       type='string', default=None, help='The path of new app bundle.')
-  parser.add_option('--output', dest='output_path', action='store',
+  parser.add_option('--new-dmg', dest='new_dmg_path', action='store',
+      type='string', default=None, help='The path of new dmg.')
+  parser.add_option('--patch-output', dest='patch_output_path', action='store',
       type='string', default=None, help='The path of generated delta file.')
+  parser.add_option('--patch-eddsa-output', dest='patch_eddsa_output_path', action='store',
+      type='string', default=None, help='The path of eddsa output of patch.')
+  parser.add_option('--dmg-eddsa-output', dest='dmg_eddsa_output_path', action='store',
+      type='string', default=None, help='The path of eddsa output of dmg.')
   (options, args) = parser.parse_args(argv)
 
   if len(args) > 0:
     print >> sys.stderr, parser.get_usage()
     return 1
 
   # generate patch file
-  command = [options.binary_delta_path, 'create', options.old_app_path, options.new_app_path, options.output_path]
+  command = [options.binary_delta_path, 'create', options.old_app_path, options.new_app_path, options.patch_output_path]
   try:
       subprocess.check_call(command)
   except subprocess.CalledProcessError as e:
       print(e.output)
       raise e
 
   # sign patch file
-  command = [options.sign_update_path, '-s', options.sign_key, options.output_path]
+  file = open(options.patch_eddsa_output_path, 'w')
+  command = [options.sign_update_path, '-s', options.sign_key, options.patch_output_path]
   try:
-      subprocess.check_call(command)
+      subprocess.check_call(command, stdout=file)
+  except subprocess.CalledProcessError as e:
+      print(e.output)
+      raise e
+  file.close()
+
+  # sign dmg
```

Comment:
we should sign the dmg even when not building the delta (if the sparkle private key is set)

Reply:
same for dsa, if the private key is passed in, we should sign the dmg